girder/api/rest.py

```diff
@@ -252,6 +253,48 @@ def setCurrentUser(user):
     cherrypy.request.girderUser = user
 
 
+def setContentDisposition(filename, disposition='attachment', setHeader=True):
+    """
+    Set the content disposition header to either inline or attachment, and
+    specify a filename that is properly escaped.  See
+    developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition,
+    tools.ietf.org/html/rfc2183, and tools.ietf.org/html/rfc6266 for
+    specificatioons and details.
+
+    :param filename: the filename to add to the content disposition header.
+    :param disposition: either 'inline' or 'attachment'.  None is the same as
+        'attachment'.  Any other value skips setting the content disposition
+        header.
+    :param setHeader: if False, return the value that would be set to the
+        Content-Disposition header, but do not set it.
+    :returns: the content-disposition header value.
+    """
+    if (not disposition or (disposition not in ('inline', 'attachment') and
+                            not disposition.startswith('form-data'))):
+        raise RestException('Error: Content-Disposition is not a recognized value.')
+    if not filename:
+        raise RestException('Error: Content-Disposition filename is empty.')
+    if not isinstance(disposition, six.binary_type):
+        disposition = disposition.encode('iso8859-1', 'ignore')
```

Comment:
Why ISO 8859-1? I couldn't find mention of it in the standard mentioned in the docstring, is it mentioned somewhere further up the RFC chain?

Reply:
Nevermind, I found it. [For posterity](https://tools.ietf.org/html/rfc5987)

Reply:
Added RFC 5987 link into the comments.

Reply:
https://tools.ietf.org/html/rfc5987#section-1